motion_predict/src/motion_predict.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "motion_predict.h"
+
+namespace Motion{
+
+cav_msgs::PredictedState MotionPredict::predictState(const geometry_msgs::Pose& pose, const geometry_msgs::Twist& twist,const double delta_t)
+{
+  Eigen::VectorXd x(4); // State Vector
+
+  x(0)=pose.position.x; // Position X
+  x(1)=pose.position.y; // Position Y
+  x(2)=twist.linear.x; // Linear Velocity X
+  x(3)=twist.linear.y; // Linear Velocity Y
+  
+  Eigen::MatrixXd F=Eigen::MatrixXd::Identity(x.size(), x.size()); // Generate identity matrix for state transition matrix
+
+  F(0,2)=delta_t; 
+  F(1,3)=delta_t;
+  
+  x = F * x; // Predict 
+
+  cav_msgs::PredictedState pobj;
+
+  pobj.predicted_position.position.x=x(0); // Predicted Position X
+  pobj.predicted_position.position.y=x(1); // Predicted Position Y
+  pobj.predicted_position.position.z=pose.position.z; // Predicted Position Z
+
+  pobj.predicted_velocity.linear.x=x(2); // Predicted Linear Velocity X
+  pobj.predicted_velocity.linear.y=x(3); // Predicted Linear Velocity Y
+  pobj.predicted_velocity.linear.z=twist.linear.z; // Predicted Linear Velocity Z
```

Comment:
You should also copy the twist.angular value here. Also above you should map the orientation field.